packages/wallet/ui/src/bridge.jsx

```diff
@@ -0,0 +1,144 @@
+// @ts-check
+/* eslint-disable import/no-extraneous-dependencies */
+import '@endo/eventual-send/shim';
+import './lockdown.js';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+Error.stackTraceLimit = Infinity;
+
+/** ISSUE: where are these defined? */
+const BridgeProtocol = /** @type {const} */ ({
+  loaded: 'walletBridgeLoaded',
+  opened: 'walletBridgeOpened',
+});
+
+/**
+ * Install a dApp "connection" where messages posted from
+ * the dApp are forwarded to localStorage and vice versa.
+ *
+ * @param {{
+ *   addEventListener: typeof window.addEventListener,
+ *   parentPost: typeof window.postMessage,
+ *   t0: number,
+ * }} io
+ */
+const installDappConnection = ({ addEventListener, parentPost, t0 }) => {
+  /** @type { string } */
+  let origin;
+
+  /** @type {(key: string, value: string) => void} */
```

Comment:
I'm surprised ts-check isn't alerting that `setItem` is undefined at this point. 

Oh, there's no type checking in this package yet. https://github.com/Agoric/agoric-sdk/pull/5757